### PR TITLE
Momenibd code review1

### PR DESCRIPTION
**Pull request by momoney13**
Description of the project:
PictureThis is a mobile application used to decide what movies to watch. The application suggests movies the user might be interested in, and then gives the user ways they can watch said movies. 

- The project was available in github on time for me to review it.
- I'd say they had enough comments and documentation on the program for me to understand how it functions properly. I rarely had any issues of not understanding how the code functioned.
- The program did compile and passed tests on Circleci. I couldn't get my emulator to run it, but that might've been due to my settings or something.

Technical Suggestions:
- I would change (1..movieList.size-1) to (1 until movieList.size). This shouldn't have any bearing on the functionality of movieList. I just think that it looks cleaner personally and conveys what it does better. https://kotlinlang.org/docs/coding-conventions.html
- I would change how your multiline comments are done. The coding conventions recommends using /** at the start of a comment, * for each sequential line, and a */ at the end to signify to the reader when a comment starts and ends. https://kotlinlang.org/docs/coding-conventions.html#documentation-comments
- I would change some of the spacing between lines to make it more readable to the user. As it stands some of the code lines are in big blocks and breaking it down into smaller portions with horizontal spacing can make it look cleaner. https://www.oracle.com/java/technologies/javase/codeconventions-whitespace.html

What I learned: 
- How to properly comment on code so that it doesn't impact the code and is perfectly readable 
- How classes with longer headers should be formatted so each constructor is a new line with indentation
- How white space in code can impact how smooth an app runs since it can buffer with too much space

Conclusion:
Overall this project looks really clean and I had some trouble finding things to fix because of that. I look forward to seeing where this project can go. (As a movie buff I would love using this)  Good luck!

three commits of mine:
https://github.com/RIA03B/ForeCast-Project/issues/5#issuecomment-864625020
https://github.com/RIA03B/ForeCast-Project/issues/5#issuecomment-864625046
https://github.com/RIA03B/ForeCast-Project/issues/5#issuecomment-864625092

### CODE REVIEW

File: app/src/main/java/DAO/tmdbApiDAO.kt [hunk 1]
```diff
@@ -7,20 +7,22 @@ import com.google.gson.GsonBuilder
 import okhttp3.*
 import java.io.IOException
 
-        //THIS FUNCTION IS THE START OF THE MOVIEAPI DTO
-        //CURRENTLY IT RETURNS 10 MOVIE OBJECTS AND PUTS THEM INTO A LIST
-        //CURRENTLY IM UNSURE HOW TO BETTER FORMAT IT AS A DAO BUT I KNOW THAT THERE ARE BETTER WAYS,,
+        /**THIS FUNCTION IS THE START OF THE MOVIEAPI DTO
```
Comment (momoney13) [inline app/src/main/java/DAO/tmdbApiDAO.kt:10]: Changed the comment format to be in line with Kotlin coding convention suggestions: https://kotlinlang.org/docs/coding-conventions.html#documentation-comments

File: app/src/main/java/DAO/tmdbApiDAO.kt [hunk 2]
```diff
@@ -7,20 +7,22 @@ import com.google.gson.GsonBuilder
...
+        */
+
         fun fetchAPI(movieList: MutableList<Movie>){
             val url = "https://api.themoviedb.org/3/movie/popular?api_key=79553149d3fbab3da78c8028a8180ce0&language=en-US&page=1"
 
             val request = Request.Builder().url(url).build()
             val client = OkHttpClient()
             client.newCall(request).enqueue(object: Callback {
-                override fun onFailure(call: Call, e: IOException) {
+                override fun onFailure(call: Call, e: IOException){
```
Comment (momoney13) [inline app/src/main/java/DAO/tmdbApiDAO.kt:21]: Changed bracket format to look cleaner

File: app/src/main/java/DAO/tmdbApiDAO.kt [hunk 3]
```diff
@@ -7,20 +7,22 @@ import com.google.gson.GsonBuilder
...
                     TODO("Not yet implemented")
                 }
 
-                override fun onResponse(call: Call, response: Response) {
+                override fun onResponse(call: Call, response: Response){
```
Comment (momoney13) [inline app/src/main/java/DAO/tmdbApiDAO.kt:25]: Changed bracket format to look cleaner

File: app/src/main/java/DAO/tmdbApiDAO.kt [hunk 4]
```diff
@@ -34,57 +36,54 @@ import java.io.IOException
...
-//        "title": "A Quiet Place Part II",
-//        "video": false,
-//        "vote_average": 7.3,
-//        "vote_count": 364
-//    },
-
 
+    public class Results(val results: List<Movie>)
 
+    /**JSON EXAMPLE
```
Comment (momoney13) [inline app/src/main/java/DAO/tmdbApiDAO.kt:46]: Similar comment change to the last one: https://kotlinlang.org/docs/coding-conventions.html#documentation-comments

File: app/src/main/java/DAO/tmdbApiDAO.kt (hunk 2 above)
Comment (discospiff) [reply app/src/main/java/DAO/tmdbApiDAO.kt:21]: For reformatting... 
1) A caution.  Significant reformats can make future commit diffs difficult, as git has to deal with significant reordering of the logic.
2) The IDE can do this automatically, and you can provide a template for the IDE to use.  Ensure all developers are using this same template, or the one that comes by default with the IDE, and have them reformat before committing.  This ameliorates issue 1, above.

File: app/src/main/java/DAO/tmdbApiDAO.kt (hunk 4 above)
Comment (discospiff) [reply app/src/main/java/DAO/tmdbApiDAO.kt:46]: If it's commented out, does it need to be here at all?

File: app/src/main/java/com/example/picturethis/MainActivity.kt [hunk 5]
```diff
@@ -20,13 +20,15 @@ class MainActivity : AppCompatActivity(){
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
-        //GETS 10 MOVIE OBJECTS FROM THE TMDB API
+
+        /**GETS 10 MOVIE OBJECTS FROM THE TMDB API*/
```
Comment (discospiff) [inline app/src/main/java/com/example/picturethis/MainActivity.kt:24]: Single line comment is fine here.  This isn't true JavaDoc or KDoc.

**Comment by discospiff** [comment]
While there are some good suggestions, many of the changes can be managed by the IDE (like autoformatting) in lieu of merging the branch.

Also, look at a bigger picture around commented out code.   Committing commented out code is considered bad practice in version control, because version control offers better alternatives:
1) Use a feature branch for experimental features.
2) Delete what is not needed.  You can always compare with history to see what used to be there.
Commented out code is considered clutter in a version control system.

**Pull request closed without merging**